pkgs/development/compilers/flutter/update.py

```diff
@@ -0,0 +1,359 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i python3 -p python3Packages.pyyaml
+
+import shutil
+import json
+import urllib.request
+import tempfile
+from sys import exit
+import os
+import subprocess
+import re
+import json
+import argparse
+import yaml
+import json
+
+
+NIXPKGS_ROOT = subprocess.Popen(['git', 'rev-parse', '--show-toplevel'],
+                                stdout=subprocess.PIPE).communicate()[0].rstrip().decode('utf-8')
+
+
+def object_to_nix(obj):
+    lines = re.sub(r'"(.*?)"(?= \=)', r'\1', json.dumps(obj,
+                   indent=2, separators=(";", " = "))).splitlines()
+    for i in range(len(lines)):
+        line = lines[i]
+        if line.strip().endswith('"') or line.strip().endswith(']') or line.strip().endswith('}'):
+            lines[i] = line.rstrip() + ';'
+    return '\n'.join(lines)[:-1]
+
+
+# Return out paths
+def nix_build(code):
+    temp = tempfile.NamedTemporaryFile(mode='w')
+    temp.write(code)
+    temp.flush()
+    os.fsync(temp.fileno())
+
+    process = subprocess.Popen(
+        ["nix", "build", "--impure", "--print-out-paths", "--no-link", "--expr",
+            f"with import {NIXPKGS_ROOT} {{}}; callPackage {temp.name} {{}}"],
+        stdout=subprocess.PIPE,
+        text=None
+    )
+
+    process.wait()
+    temp.close()
+    return process.stdout.read().decode('utf-8').strip().splitlines()[0]
+
+
+# Return errors
+def nix_build_to_fail(code):
+    temp = tempfile.NamedTemporaryFile(mode='w')
+    temp.write(code)
+    temp.flush()
+    os.fsync(temp.fileno())
+
+    process = subprocess.Popen(
+        ["nix", "build", "--impure", "--keep-going", "--no-link", "--expr",
+            f"with import {NIXPKGS_ROOT} {{}}; callPackage {temp.name} {{}}"],
+        stderr=subprocess.PIPE,
+        text=None
+    )
+
+    stderr = ""
+    while True:
+        line = process.stderr.readline()
+        if not line:
+            break
```

Comment:
`line = line.decode('utf-8')` so you decode all once

Reply:
True
Though I ended up just using `text=True`